src/main/scala/learnshapeless/HLists.scala

```diff
@@ -32,17 +32,23 @@ object HLists extends App {
     (a) start the `sbt console`, and enter `:type learnshapeless.HLists.ex_newton`
     (b) if you are using Intellij, you can focus cursor on the expression and activate TypeInfo command (Ctrl-Shift-P on my Mac)
      */
-  def ex_newton = ???
+  def ex_newton: String :: Int :: Country :: HNil =
+    "Newton" :: 1642 :: England :: HNil
+  println(s"ex_newton $ex_newton")
```

Comment:
Perhaps you can preface the exercise with the problem `HList` is trying to solve?
* No type safety checking for `Seq` of a mixed types (because it now becomes `Seq[Any]`)
* No type safety checking for `Seq` of same types of a known length

After doing this exercise, it feels to me that `HList` is more powerful tuple, but the advantage alone will not led me to immediately reached for it.

Perhaps `HList` is for people building libraries, frameworks, and maybe DSL? I can imagine myself combining tagged types, `HList` and `Poly` to build a validation framework, db access (like Slick), or parser of some sort.